v5/core/base_service.go

```diff
@@ -348,25 +391,9 @@ func (service *BaseService) Request(req *http.Request, result interface{}) (deta
 		}
 	}
 
+	// Invoke the request, then check for errors during the invocation.
 	var httpResponse *http.Response
-
```

Comment:
While trapsing through the "retryablehttp" package source code to learn what I needed to fix the "disable ssl" problem, I discovered that we do not need to distinguish between retryable and non-retryable clients at this level.  This is because when retries are enabled, the BaseService.Client will hold a "shim" client instance whose "Do" method can be called just like the non-retryable scenario and the retry logic will get triggered as expected.